public/scripts/ebayCall.js

```diff
@@ -0,0 +1,65 @@
+// Parse the response and build an HTML table to display search results
+// function _cb_findItemsByKeywords(root) {
+function _cb_findItemsAdvanced  (root) {
+  var items = root.findItemsAdvancedResponse[0].searchResult[0].item || [];
+  var html = [];
+  html.push('<table width="75%" border="0" cellspacing="0" cellpadding="3" id="ebayTable"><tbody>');
+  for (var i = 0; i < items.length; ++i) {
+    var item     = items[i];
+    var title    = item.title;
+    var pic      = item.galleryURL;
+    var viewitem = item.viewItemURL;
+    if (null != title && null != viewitem) {
+      html.push('<tr><td>' + '<img src="' + pic + '" border="0">' + '</td>' +
+      '<td><a href="' + viewitem + '" target="_blank">' + title + '</a></td></tr>');
+    }
+  }
+  html.push('</tbody></table>');
+  $('#ebay').append(html.join(""));
+}
+
+// Create a JavaScript array of the item filters you want to use in your request
+var filterarray = [
+  {"name":"MaxPrice",
+   "value":"10000",
+   "paramName":"Currency",
+   "paramValue":"USD"},
+  {"name":"ListingType",
+   "value": ["AuctionWithBIN", "FixedPrice", "StoreInventory"],
+   "paramName":"",
+   "paramValue":""},
+];
+
+// Define global variable for the URL filter
+var urlfilter = "";
+
+// Generates an indexed URL snippet from the array of item filters
+function  buildURLArray() {
+  // Iterate through each filter in the array
+  for(var i=0; i<filterarray.length; i++) {
+    //Index each item filter in filterarray
+    var itemfilter = filterarray[i];
+    // Iterate through each parameter in each item filter
+    for(var index in itemfilter) {
+      // Check to see if the paramter has a value (some don't)
+      if (itemfilter[index] !== "") {
+        if (itemfilter[index] instanceof Array) {
+          for(var r=0; r<itemfilter[index].length; r++) {
+            var value = itemfilter[index][r];
+            urlfilter += "&itemFilter\(" + i + "\)." + index + "\(" + r + "\)=" + value ;
+          }
+        }
+        else {
+          urlfilter += "&itemFilter\(" + i + "\)." + index + "=" + itemfilter[index];
+        }
+      }
+    }
+  }
+}
+
+// Execute the function to build the URL filter
+buildURLArray(filterarray);
+
+// Declare global url for building up ebay GET call query string
+// This reates unique url for each artist as they pass through a for loop, in ebayAuction.js
+var url; //
```

Comment:
`var url` is defined and used again in ebayAuction.js within your for loop to build the URL. Is it necessary to have it in both places?